File: python/findMiddleNumber/naive.py
class Solution:
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        len1, len2 = len(nums1), len(nums2)
        totalLen = len1 + len2
        middle = totalLen // 2
        i, j, a, isFound = 0, 0, [0, 0], False
        while i < len1 and j < len2:
            if nums1[i] < nums2[j]:
                a[0] = a[1]
                a[1] = nums1[i]
                i += 1
            else:
                a[0] = a[1]
                a[1] = nums2[j]
                j += 1
            if i + j > middle:
                isFound = True
                break
        if not isFound:
            if i + j == middle:
                a[0], a[1] = a[1], nums1[i] if j >= len2 else nums2[j]
            elif i >= len1:
                a[0], a[1] = nums2[middle - i - 1], nums2[middle - i]
            else:
                a[0], a[1] = nums1[middle - j - 1], nums1[middle - j]
            print('i=%d, j=%d, a=%d %d' % (i, j, a[0], a[1]))
        if totalLen % 2:
            return a[1]
        return (a[0] + a[1]) / 2


if __name__ == '__main__':
    sol = Solution()
    a = [1, 2, 3]
    b = [4, 5, 6]
    print(sol.findMedianSortedArrays(a, b))
    a = [1, 2]
    b = [2.5, 5, 6, 7, 8]
    print(sol.findMedianSortedArrays(a, b))
    a = [1, 2, 3]
    b = [4, 5]
    print(sol.findMedianSortedArrays(a, b))
    a = [2]
    b = [1, 3, 4]
    print(sol.findMedianSortedArrays(a, b))

File: old/C++/Array/findMedianSortedArray/naive.cpp
#include <vector>
using namespace std;

// 如果直接用double a数组的话，从int到double的转换会消耗大量时间；
// 本来的排名是战胜60%左右；
// 改成int，并在最后转double，这样的排名是战胜了95%左右
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int a[] = {0, 0};
        int i = 0, j = 0, len1 = nums1.size(), len2 = nums2.size();
        int middle = (len1 + len2) / 2;
        while(i < len1 || j < len2) {
            a[0] = a[1];
            if(i == len1) a[1] = nums2[j++];
            else if(j == len2) a[1] = nums1[i++];
            else if(nums1[i] < nums2[j]) a[1] = nums1[i++];
            else a[1] = nums2[j++];
            if(i + j > middle) break;
        }
        if((len1 + len2) % 2) return double(a[1]);
        else return (a[0] + a[1]) / 2.0;
    }
};

int main() {
    vector<int> a, b;
    a.push_back(1);
    a.push_back(3);
    b.push_back(2);
    printf("%lf", Solution().findMedianSortedArrays(a, b));
}